### PR TITLE
Updates the fiat fork to 3.0.2, fiat 0.1.6

### PR DESCRIPTION
**Pull request by huitseeker**
In order for this to lead to a dependency update of diem with backward-compatibility, this needs to be merged to a new branch "fiat4" on the novifinancial/curve25519-dalek repo (set initially to track dalek-cryptography/curve25519-dalek@master - NOTE: I can't create this branch on my own)

I'll open a PR to update novifinancial/curve25519@master to the latest dalek-cryptography/curve25519-dalek@master in a distinct PR (edit: ready to merge at #13 ). Once that one is merged, all that's left to do is to push a "fiat4"branch at that updated tag by somebody with write rights.

/cc @kchalkias @rexhoffman for review and branch creation

### CODE REVIEW

File: src/backend/vector/avx2/edwards.rs
Comment (kchalkias) [inline src/backend/vector/avx2/edwards.rs]: this was just a comment issue or the logic actually changed?

File: .travis.yml
Comment (kchalkias) [inline .travis.yml]: so the avx2 backend is deprecated by the simd one or we should support both?

**Comment by huitseeker** [comment]
@kchalkias thanks for the prompt attention. Please do not review changes that are part of the Dalek upgrade 3.0.1 -> 3.0.2 here, head on to #13 instead. I'd like to keep this present PR for the fiat_u64_backend and fiat-crypto update.

File: .travis.yml
Comment (huitseeker) [reply .travis.yml]: It's a refactoring to avoid conflicts on some AMD processors:
https://github.com/dalek-cryptography/curve25519-dalek/pull/334
We've never used either directly.

File: src/backend/vector/avx2/edwards.rs
Comment (huitseeker) [reply src/backend/vector/avx2/edwards.rs]: The logic did not change ( https://github.com/dalek-cryptography/curve25519-dalek/commit/8da05f7e90b59b6dea6f72c4d4cad238647b1b60#diff-f507d38947f74ad1450f16fa341b60ddb21d9d77b7db0cb700fac4fa3c783eeb )

**Comment by rexhoffman** [comment]
@kchalkias assuming you have write access and will leave this to you.   @huitseeker I'm tempted to move this repo to diem org so we can give you write access.

**Comment by huitseeker** [comment]
@kchalkias 
```bash
cd checkout/of/curve25519-dalek
NOVI_FORK=$(git remote -v| grep 'github\.com/novifinancial/curve25519-dalek'| awk '{print $1}'|uniq)
git checkout master && git reset --hard $NOVI_FORK/master
git checkout -b fiat4
git push -u $NOVI_FORK fiat4
```

and then I'll rebase on `fiat4`.

**Comment by huitseeker** [comment]
@kchalkias Good to go! upcoming PR on diem to adopt this.

**Review by kchalkias — APPROVED** [review]
LGTM

**Pull request merged**
